app/frontend/js/apps/TransformationApp/JumpTo.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { askNewRawRecord, selectRawRecord } from "/js/features/RawRecordSlice";
+import { selectAppDetails } from "/js/features/AppDetailsSlice";
+
+const JumpTo = () => {
+  const dispatch = useDispatch();
+
+  const { transformationDefinition } = useSelector(selectAppDetails);
+
+  const { isFetching, page, record, totalPages, totalRecords } =
+    useSelector(selectRawRecord);
+
+  const [inputPage, setInputPage] = useState(page);
+  const [inputRecord, setInputRecord] = useState(record);
+
+  useEffect(() => {
+    setInputPage(page);
+    setInputRecord(record);
+  }, [page, record]);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(
+      askNewRawRecord({
+        transformationDefinitionId: transformationDefinition.id,
+        page: inputPage,
+        record: inputRecord,
+      })
+    );
+  };
+
+  const handleNextRecordClick = (e) => {
+    dispatch(
+      askNewRawRecord({
+        transformationDefinitionId: transformationDefinition.id,
+        page: record === totalRecords ? page + 1 : page,
+        record: record === totalRecords ? 1 : record + 1,
+      })
+    );
+  };
+
+  const handlePreviousRecordClick = (e) => {
+    dispatch(
+      askNewRawRecord({
+        transformationDefinitionId: transformationDefinition.id,
+        page: record === 1 ? page - 1 : page,
+        record: record === 1 ? totalRecords : record - 1,
+      })
+    );
+  };
+
+  return (
+    <div className="dropdown jump-to">
+      <button
+        className="btn btn-primary dropdown-toggle text-truncate"
+        type="button"
+        data-bs-toggle="dropdown"
+        data-bs-auto-close="outside"
+        aria-expanded="false"
+      >
+        Jump to
+      </button>
+      <div className="dropdown-menu dropdown-menu-end jump-to__body p-2">
+        <form className="row gx-1 mb-3" onSubmit={handleSubmit}>
+          <label className="col-3 col-form-label" htmlFor="page">
+            Page
+          </label>
+          <div className="col">
+            <div className="input-group">
+              <input
+                name="page"
+                id="page"
+                className="form-control"
+                disabled={isFetching || record !== inputRecord}
+                type="number"
+                value={inputPage}
+                onChange={(e) => setInputPage(parseInt(e.target.value))}
+              />
+              <span className="input-group-text">/ {totalPages}</span>
+            </div>
+          </div>
+          <div className="col-auto">
+            <button
+              className="btn btn-primary"
+              disabled={
```

Comment:
Maybe put this logic in a clearly named method so you can see disabled={method} rather than all these conditions.

Reply:
Done :)